contracts/allowlist/ValidatorSmartContractAllowList.sol

```diff
@@ -0,0 +1,127 @@
+// Implementation of a contract to select validators using an allowlist
+
+pragma solidity >=0.5.0;
+
+import "../ValidatorSmartContractInterface.sol";
+
+contract ValidatorSmartContractAllowList is ValidatorSmartContractInterface {
+
+    struct accountInfo {
+        bool allowed;
+        bool activeValidator;
+        uint8 validatorIndex;
+    }
+
+    address[] private validators;
+    mapping(address => accountInfo) private allowedAccounts;
+    uint public numAllowedAccounts;
+    mapping(address => address[]) private currentVotes;
+
+    modifier senderIsAllowed() {
+        require(allowedAccounts[msg.sender].allowed == true, "sender is not on the allow list");
+        _;
+    }
+
+    constructor (address[] memory initialAccounts, address[] memory initialValidators) public {
+        require(initialAccounts.length > 0, "no inital allowed accounts");
+        require(initialValidators.length > 0, "no inital validator accounts");
+
+        require(initialAccounts.length >= initialValidators.length, "number of initial accounts smaller than number of inital validators");
+        for (uint i = 0; i < initialAccounts.length; i++) {
+            require(initialAccounts[i] != address(0), "initial accounts cannot be zero");
+            numAllowedAccounts++;
```

Comment:
I am not sure if this is optimized into a single store. It is probably better to just assign `numAllowedAccounts = initialAccounts.length` outside of the loop.